### PR TITLE
Refactor ModulesList component and add stories for the same.

### PR DESCRIPTION
**Pull request by kostyalmm**
## Summary

Addresses issue #2194

## Relevant technical choices

* Refactor ModulesList component to functional component
* Add story for the same
* Update imports

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: package-lock.json
Comment (eugene-manuilov) [inline package-lock.json:8518]: @kostyalmm seems like `package-lock.json` has been updated without dependencies change in the `package.json` file. Could you please revert changes in the `package-lock.json`?

File: assets/js/components/ModulesList.js
Comment (eugene-manuilov) [inline assets/js/components/ModulesList.js:4]:
```suggestion
 * Site Kit by Google, Copyright 2020 Google LLC
```

File: assets/js/components/ModulesList.js
Comment (eugene-manuilov) [inline assets/js/components/ModulesList.js:22]:
```suggestion
import map from 'lodash/map';
```

File: assets/js/components/ModulesList.js
Comment (eugene-manuilov) [inline assets/js/components/ModulesList.js:73]: @kostyalmm it misses the following from the IB:

```
Call the trackEvent function like its done in the activateOrDeactivateModule function;
```

File: assets/js/components/ModulesList.js
Comment (eugene-manuilov) [inline assets/js/components/ModulesList.js:87]: Modules, received from the `core/modules` datastore, don't have `setupComplete` property. Use `connected` instead.

File: assets/js/components/ModulesList.js
Comment (eugene-manuilov) [inline assets/js/components/ModulesList.js:107]: `active` means that the module is active but not connected (a user hasn't configured it yet). We need to use the `connected` property to determine if the module connected or not.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Hi @kostyalmm! Thanks for implementing it. I have found some issues that need to be addressed before we can merge it. Could you please address it?

File: assets/js/components/ModulesList.js
Comment (eugene-manuilov) [inline assets/js/components/ModulesList.js:113]: @kostyalmm  sorry for the incorrect suggestion at the beginning, but here we need to rely on the `active` property instead of `connected`.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @kostyalmm! Mostly LGTM. Just found one thing to change.

File: assets/js/components/ModulesList.js
Comment (kostyalmm) [reply assets/js/components/ModulesList.js:113]: Are you sure? 🤔 `connected` seems to be the correct property here according to me as the [old component has](https://github.com/google/site-kit-wp/blob/develop/assets/js/components/modules-list.js#L119):
```
setupComplete: isConnected,
```

By this logic, `connected` seems like the right property. What do you think?

Comment (eugene-manuilov) [reply assets/js/components/ModulesList.js:113]: @kostyalmm as far as I understand it, `connected` means that the module is activated and the user has configured which is not what we need here. We need to know whether a module is activated and if it's not, let the user activate it by clicking on the "Connect" link.

@aaemnnosttv am I right?

Comment (aaemnnosttv) [reply assets/js/components/ModulesList.js:113]: @eugene-manuilov `setupComplete` for the legacy module format is equal to `module.active && module.connected`:
https://github.com/google/site-kit-wp/blob/27d4df7bbfbf6c7e59310ee9338e2a6155870268/includes/Core/Modules/Modules.php#L159

So in this case, it's not a 1-to-1 equivalent logically but it is the closest thing for `setupComplete`.

With that said, the consuming logic would _also_ need to check for `active` as well whereas before it was included in the definition of `setupComplete`.

cc @kostyalmm

Comment (eugene-manuilov) [reply assets/js/components/ModulesList.js:113]: Thanks, @aaemnnosttv! @kostyalmm let's add `active` and use `( active && connected )` to determine whether a module is connected or not.

**Review by eugene-manuilov — APPROVED** [review]
LGTM! Thanks, @kostyalmm!

File: stories/modules-list.stories.js
Comment (aaemnnosttv) [inline stories/modules-list.stories.js:35]: Please use the `provideModules` utility instead.

File: stories/notifications.stories.js
Comment (aaemnnosttv) [inline stories/notifications.stories.js:24]: See above.

File: stories/notifications.stories.js
Comment (aaemnnosttv) [inline stories/notifications.stories.js:33]: Please update the URL, using an imported image to be consistent with other stories as this recently changed.

File: stories/notifications.stories.js
Comment (aaemnnosttv) [inline stories/notifications.stories.js:32]: Please remove as this isn't a prop of `Notification`.

File: assets/js/components/ModulesList.js
Comment (aaemnnosttv) [inline assets/js/components/ModulesList.js:61]: The `try/catch` here is no longer working as it was before as neither `activateModule` or `trackEvent` will throw. Instead, `error` should come from the return value of `activateModule`, just as it does with our other async actions.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @kostyalmm ! There's still a few changes to make here as the new component is still too close to the old implementation. In particular, there is a substantial amount of code to remove/update related to the module dependencies which was to be removed per the ACs.

Regarding the changes to the VRT reference images – the new files look identical, can you explain why this was necessary? The refactoring here shouldn't have required updating these images.

File: assets/js/components/ModulesList.js
Comment (kostyalmm) [reply assets/js/components/ModulesList.js]: I removed the filter but I kept the `blockedByParentModule `. Let me explain why:
As I understand, the `ModulesList` component will show a list of modules that can be activated. The modules which can't be activated because their dependency is not activated, should show a greyed out message asking to activate the dependency first. 

I made a few changes to how `blockedByParentModule` was set and the surrounding code. 

I'd like to request you to pull the latest changes, start storybook and go to the `ModulesList` story. Once you go you will see all the modules including `TagManager`  and `Optimize` since `provideModules` activates `Analytics` by default.
Now go to `fixtures.json` and set `active` for `Analytics` to false, wait for storybook to reload, and you will see that `TagManager` and `Optimize` can't be activated and shows the desired message. 

Do let me know if this is the desired requirement for this Component. If not, then I will remove `blockedByParentModule ` as you suggested above. In that case I have two questions:
1. Will the modules with *unactivated* dependencies be shown by default?
2. Will the aforementioned modules show the greyed out message asking to activate `dependency`? Or will they be allowed to be activated unconditionally?
3. Will the aforementioned modules be not shown?

File: stories/notifications.stories.js
Comment (kostyalmm) [reply stories/notifications.stories.js:30]: It seems that image imports can't be done like 
```js
winImage={ global._googlesitekitLegacyData.admin.assetsRoot + rocketImage }
```

as it results in a src 
```
http://localhost:9001/assets/static/media/rocket.3e64a093.png
```
which gives a 404.

Am I doing something wrong here? The other image files are referenced directly by the file name in the same `Notifications` story scope.

**Comment by kostyalmm** [comment]
Regarding the VRT changes, the change is there because earlier the `ModulesList` component was using `getModulesData` which in turn used `global._googlesitekitLegacyData`, this has been now changed to use the datastore. In the changes before the requested changes were made by @aaemnnosttv , the stories were supplied with `FIXTURES` which had different data.

After making the requested changes, there are still difference in the VRT because it now shows all the modules rather than showing the ones which don't have any dependency. <= See [this comment](https://github.com/google/site-kit-wp/pull/2375#discussion_r528721724) for more info

File: stories/notifications.stories.js
Comment (aaemnnosttv) [reply stories/notifications.stories.js:30]: Oh sorry, my mistake. This changed recently in the PR for #2043 and it just hasn't been merged yet but I thought it had. You're right, this should be as you have it now 👍

File: assets/js/components/ModulesList.js
Comment (aaemnnosttv) [reply assets/js/components/ModulesList.js]: Regarding my comment above, I wasn't suggesting we remove the filter, I was only suggesting we update that part (as it's relevant to the ACs) to be a bit more clear either by using separate variables or inline it perhaps to not have a variable called "sorted" which also filters. 

Thanks for the additional context. It's important to understand that the issue was defined specifically with the removal of `blockedByParentModule` behavior as mentioned in the AC because it is not possible to work otherwise with the current/legacy version of the component. To do otherwise means that we would be introducing a behavioral change to the component where it is not called for and not currently present. 

While this may have been a bug and I can appreciate your desire to fix it, a conflict with the AC should be raised with @felixarntz as soon as possible. As it is, I think the issue is clear about what should be done but we can certainly wait for @felixarntz to give the final word.

File: .storybook/storybook-data.js
Comment (aaemnnosttv) [inline .storybook/storybook-data.js:180]: This was probably a mistake but please don't change other VRT scenarios.

File: .storybook/storybook-data.js
Comment (aaemnnosttv) [inline .storybook/storybook-data.js:174]: This issue doesn't call for a new VRT scenario to be added since this component is already covered by other existing scenarios, it should only have the added Storybook story.

File: package-lock.json
Comment (aaemnnosttv) [inline package-lock.json:10837]: Please don't include changes to the `package-lock.json` for issues which don't include dependency changes.

**Review by aaemnnosttv — COMMENTED** [review]
> After making the requested changes, there are still difference in the VRT because it now shows all the modules rather than showing the ones which don't have any dependency.

I believe it should continue to only show modules which have no dependencies as before but let's wait for @felixarntz to confirm. See [my comment](https://github.com/google/site-kit-wp/pull/2375#discussion_r528823006) on the thread above. There should be no visual changes needed to the existing scenarios let's see how to reconcile those rather than updating references and remove the added scenario.

File: assets/js/components/ModulesList.js
Comment (felixarntz) [reply assets/js/components/ModulesList.js]: @kostyalmm @aaemnnosttv Indeed the ability of showing modules with inactive dependencies greyed out is legacy behavior, which used to be the case, but was intentionally changed a while ago. Unfortunately the code back then wasn't fully removed which is probably what created the confusion here. The way things currently look after your latest commits should be good.

File: assets/js/components/ModulesList.js
Comment (aaemnnosttv) [inline assets/js/components/ModulesList.js:90]: This comment is no longer relevant.
```suggestion
```

**Review by aaemnnosttv — COMMENTED** [review]
LGTM, just one tiny thing to clean up from the old component.

File: assets/js/components/ModulesList.js
Comment (aaemnnosttv) [inline assets/js/components/ModulesList.js:93]:
```suggestion
	const modules = Object.values( moduleObjects )
		.filter( ( module ) => ! module.internal && 0 === module.dependencies.length )
		.sort( ( module1, module2 ) => module1.sort - module2.sort );

	return (
		<div className="googlesitekit-modules-list">
			{ modules.map( ( module ) => {
```

**Review by aaemnnosttv — APPROVED** [review]
LGTM 👍

**Pull request merged**
